src/test/java/com/google/cloud/tools/gradle/appengine/core/DeployAllTaskTest.java

```diff
@@ -56,31 +54,30 @@
 
   private DeployAllTask deployAllTask;
 
+  private File stageDir;
```

Comment:
Do we have a test to ensure the main config isn't changed by this task?

Reply:
I think that'll need to go somewhere else. It does change what's passed in, we just pass in a copy of the config when the task is created, so I can't really check that in the unit tests.

Reply:
Perhaps a safer pattern would be to pass around an immutable `DeployConfiguration` built from `DeployExtension` rather than rely on `#setDeployConfig` to be called with a copy of the mutable `DeployExtension`, but that might be overkill. A simple fix would probably be just moving the `new DeployExtension` call to inside `#setDeployConfig`.

Reply:
I think the reason I didn't do that originally was so I could mock the internal `deployConfig` in tests.

Reply:
Maybe we should go with a clone() method instead of a copy constructor?

Reply:
One way you could do it is to capture and assert on the `deployConfig` passed into `deploy.deploy` and not use a mocked `DeployExtension` since it doesn't have any side effects.

Reply:
@TadCordle oh right I forgot that we put in the config step. Yeah, the no side effects would be ideal. I didn't think of the capture though, that might be the way we want to go.